python/ray/tune/logger/aim.py

```diff
@@ -0,0 +1,209 @@
+            if k in tmp_result:
+                del tmp_result[k]  # not useful to log these
+
+        context = None
+        if "context" in tmp_result:
+            context = tmp_result["context"]
+            del tmp_result["context"]
+
+        epoch = None
+        if "epoch" in tmp_result:
+            epoch = tmp_result["epoch"]
+            del tmp_result["epoch"]
+
+        if self._metrics:
+            for metric in self._metrics:
+                try:
+                    self._trial_run[trial].track(
+                        value=tmp_result[metric],
+                        epoch=epoch,
+                        name=metric,
+                        step=step,
+                        context=context,
+                    )
+                except KeyError:
+                    logger.warning(
+                        f"The metric {metric} is specified but not reported."
+                    )
+        else:
+            # if no metric is specified log everything that is reported
+            flat_result = flatten_dict(tmp_result, delimiter="/")
+            valid_result = {}
+
+            for attr, value in flat_result.items():
+                if isinstance(value, tuple(VALID_SUMMARY_TYPES)) and not np.isnan(
+                    value
+                ):
+                    valid_result[attr] = value
+                    self._trial_run[trial].track(
+                        value=value, name=attr, epoch=epoch, step=step, context=context
+                    )
+                elif (isinstance(value, list) and len(value) > 0) or (
+                    isinstance(value, np.ndarray) and value.size > 0
+                ):
+                    valid_result[attr] = value
+
+                    """
+                    # ToDO: implement usage of audio, images etc. here..
+                    # Must be video
+                    if isinstance(value, np.ndarray) and value.ndim == 5:
+                        self._trial_writer[trial].add_video(
+                            attr, value, global_step=step, fps=20
+                        )
+                        continue
+
+                    try:
+                        self._trial_writer[trial].add_histogram(
+                            attr, value, global_step=step
+                        )
+                    # In case TensorboardX still doesn't think it's a valid value
+                    # (e.g. `[[]]`), warn and move on.
+                    except (ValueError, TypeError):
+                        if log_once("invalid_tbx_value"):
+                            logger.warning(
+                                "You are trying to log an invalid value ({}={}) "
+                                "via {}!".format(attr, value, type(self).__name__)
+                            )
+                    """
+
+    def log_trial_end(self, trial: "Trial", failed: bool = False):
+        # cleanup in the end
+        self._trial_run[trial].finalize()
```

Comment:
The call of `Run.finalize()` is not needed. `Run.close()` is the method to finalize the Run and release all resources.